index.d.ts

```diff
@@ -0,0 +1,373 @@
+   *
+   * @default undefined
+   */
+  importer?: Importer | Importer[];
+
+  /**
+   * Holds a collection of custom functions that may be invoked by the sass files being compiled.
+   *
+   * @default undefined
+   */
+  functions?: {
+    [key: string]: (...args: types.SassType[]) => types.SassType | void;
+  };
+
+  /**
+   * An array of paths that should be looked in to attempt to resolve your @import declarations.
+   * When using `data`, it is recommended that you use this.
+   *
+   * @default []
+   */
+  includePaths?: string[];
+
+  /**
+   * Enable Sass Indented Syntax for parsing the data string or file.
+   *
+   * @default false
+   */
+  indentedSyntax?: boolean;
+
+  /**
+   * Used to determine whether to use space or tab character for indentation.
+   *
+   * @default 'space'
+   */
+  indentType?: 'space' | 'tab';
+
+  /**
+   * Used to determine the number of spaces or tabs to be used for indentation.
+   *
+   * @default 2
+   */
+  indentWidth?: number;
+
+  /**
+   * Used to determine which sequence to use for line breaks.
+   *
+   * @default 'lf'
+   */
+  linefeed?: 'cr' | 'crlf' | 'lf' | 'lfcr';
+
+  /**
+   * Disable the inclusion of source map information in the output file.
+   *
+   * @default false
+   */
+  omitSourceMapUrl?: boolean;
+
+  /**
+   * Specify the intended location of the output file.
+   * Strongly recommended when outputting source maps so that they can properly refer back to their intended files.
+   *
+   * @default null
+   */
+  outFile?: string;
+
+  /**
+   * Determines the output format of the final CSS style.
+   *
+   * @default 'expanded'
+   */
+  outputStyle?: 'compressed' | 'expanded';
+
+  /**
+   * Enables the outputting of a source map.
+   *
+   * @default undefined
+   */
+  sourceMap?: boolean | string;
+
+  /**
+   * Includes the contents in the source map information.
+   *
+   * @default false
+   */
+  sourceMapContents?: boolean;
+
+  /**
+   * Embeds the source map as a data URI.
+   *
+   * @default false
+   */
+  sourceMapEmbed?: boolean;
+
+  /**
+   * The value will be emitted as `sourceRoot` in the source map information.
+   *
+   * @default undefined
+   */
+  sourceMapRoot?: string;
+
+  /**
+   * Callbacks for handling any `@warn` or `@debug` messages
+   *
+   * @default undefined
+   */
+  logger?: Logger;
+}
+
+export interface SassException extends Error {
+  /**
+   * The error message.
+   */
+  message: string;
+
+  /**
+   * The formatted error.
+   */
+  formatted: string;
+
+  /**
+   * The line number of error.
+   */
+  line: number;
+
+  /**
+   * The column number of error.
+   */
+  column: number;
+
+  /**
+   * The status code.
+   */
+  status: number;
+
+  /**
+   * The filename of error.
+   *
+   * In case file option was not set (in favour of `data`), this will reflect the value `stdin`.
+   */
+  file: string;
+}
+
+export interface SourceLocation {
```

Comment:
Like the `Logger` classes, these source interfaces should be thoroughly specified.